### PR TITLE
scheduler: enlarge the search space so that the hot-scheduler can still schedule under dimensional conflicts

### PR DESCRIPTION
**Pull request by lhy1024**
### What problem does this PR solve?

Issue Number: Ref https://github.com/tikv/pd/issues/4949

### What is changed and how does it work?

1. enlarge the search space so that the hot-scheduler can still schedule under dimensional conflicts
2. Introduce stddev to reduce unnecessary scheduling

### Check List

Tests

- Unit test
- Integration test
- Manual test (add detailed scripts or steps below)

#### master
image

#### master with close strcit-pick
image

#### pr
image

### Release note

```release-note
enlarge the search space so that the hot-scheduler can still schedule under dimensional conflicts
```

### CODE REVIEW

File: server/schedulers/hot_region.go [hunk 1]
```diff
@@ -473,6 +481,12 @@ func (bs *balanceSolver) solve() []*operator.Operator {
 	}
 	bs.cur = &solution{}
 	tryUpdateBestSolution := func() {
+		if bs.cur.progressiveRank == -1 && bs.isUniformFirstPriority(bs.cur.srcStore) {
```
Comment (HunDunDM) [inline server/schedulers/hot_region.go:484]: `bs.isUniformFirstPriority(bs.cur.srcStore)` can be computed only once after `for _, srcStore := range bs.filterSrcStores()`?

**Review by HunDunDM — APPROVED** [review]

File: server/schedulers/hot_region.go [hunk 2]
```diff
@@ -486,7 +501,11 @@ func (bs *balanceSolver) solve() []*operator.Operator {
 	for _, srcStore := range bs.filterSrcStores() {
 		bs.cur.srcStore = srcStore
 		srcStoreID := srcStore.GetID()
-
+		isUniformFirstPriority, isUniformSecondPriority := bs.isUniformFirstPriority(bs.cur.srcStore), bs.isUniformSecondPriority(bs.cur.srcStore)
+		if isUniformFirstPriority && isUniformSecondPriority {
```
Comment (nolouch) [inline server/schedulers/hot_region.go:505]: Do you test multi-workloads running on one cluster? like `sysbench` + `tpcc`

Comment (nolouch) [inline server/schedulers/hot_region.go:505]: Do you test multi-workloads running on one cluster? like `sysbench` + `tpcc`

**Review by nolouch — APPROVED** [review]

**Pull request merged**